src/goose/executor.clj

```diff
@@ -3,30 +3,43 @@
     [goose.defaults :as d]
     [goose.redis :as r]
     [goose.retry :as retry]
+    [goose.statsd.statsd :as statsd]
     [goose.utils :as u]
 
     [clojure.tools.logging :as log]))
 
+(defn- calculate-latency
+  [job]
+  (cond
+    (:retry-at (:state job))
+    [statsd/retry-latency (- (u/epoch-time-ms) (:retry-at (:state job)))]
+    (:schedule job)
+    [statsd/schedule-latency (- (u/epoch-time-ms) (:schedule job))]
+    :else
+    [statsd/execution-latency (- (u/epoch-time-ms) (:enqueued-at job))]))
```

Comment:
Does the actual latency calculation: `(- (u/epoch-time-ms) <a time depending on job status>)` fit better in the `job` namespace? The statsd namespace can have a function to return the right label depending on the job status as well. This function ties together the calculation and finding the statsd label.

Reply:
done